sky/execution.py

```diff
@@ -112,6 +112,7 @@ def _execute(
     # pylint: disable=invalid-name
     _is_launched_by_jobs_controller: bool = False,
     _is_launched_by_sky_serve_controller: bool = False,
+    _is_controller: Optional[controller_utils._ControllerSpec] = None,
```

Comment:
`_is_xxx` naming makes me think this is a bool. Rename to `_controller_type`? Also, should type hint be `controller_utils.Controllers` enum?

Reply:
BTW, instead of adding this arg and updating upstream methods, might be cleaner to infer _is_controller inline by calling `controller_utils.Controllers.from_name(cluster_name)`.